pkg/unittest/validators/length_equal_document_validator.go

```diff
@@ -26,15 +26,13 @@ func (v LengthEqualDocumentsValidator) singleValidateCounts(manifest common.K8sM
 	}
 
 	specArr, ok := spec[0].([]interface{})
-	if !ok {
+	if !ok && count == nil {
 		return false, splitInfof(errorFormat, idx, fmt.Sprintf("%s is not array", path)), 0
 	}
 	specLen := len(specArr)
-	if count > -1 {
-		if specLen != count {
-			return false, splitInfof(errorFormat, idx, fmt.Sprintf(
-				"count doesn't match as expected. expected: %d actual: %d", count, specLen)), 0
-		}
+	if count != nil && *count > -1 && specLen != *count {
```

Comment:
Is the `*count > -1` necessary?

Reply:
At the moment there is no validation for count being negative. So this fix to work only when count is zero or positive

Reply:
I think you would need another error message for validation of user input, so instead of receiving `count doesn't match as expected. expected: 10 actual: -10` you would receive an `count must be greater or equal to 0 if given. Actual: -10`. 

Even better, config validation should be done before even starting the tests, so the developer immediately knows if he has made an error

Reply:
Yeah hard to disagree. I leave it for a follow-up or someone else to contribute ;-)

Reply:
Hey @AndersBennedsgaard. I've created an issue https://github.com/helm-unittest/helm-unittest/issues/368

You are welcome to contribute ;-)